slither/tools/read_storage/read_storage.py

```diff
@@ -0,0 +1,521 @@
+
+            if is_user_defined_type(type_) and is_struct(type_.type):
+                tmp[var_name]["elems"] = _all_struct_variables(
+                    var, contracts, address, rpc_url, storage_address
+                )
+                continue
+
+            if is_array(type_):
+                if type_.is_fixed_array:  # arr[i]
+                    val = int(str(type_.length))
+
+                if isinstance(val, str):  # arr[] or arr[i][]
+                    # The length of dynamic arrays is stored at the starting slot
+                    val = int(val, 16)
+
+                elems = {}
+                if is_user_defined_type(type_.type):
+                    for i in range(min(val, max_depth)):
+                        elems[i] = _all_struct_variables(
+                            var, contracts, address, rpc_url, storage_address, key=str(i)
+                        )
+                        continue
+
+                else:
+                    for i in range(min(val, max_depth)):
+                        slot, val, type_string = get_storage_slot_and_val(
+                            contracts,
+                            address,
+                            var_name,
+                            rpc_url,
+                            storage_address,
+                            key=str(i),
+                        )
+                        elems[i] = {"slot": slot, "value": val, "type_string": type_string}
+
+                        if is_array(type_.type):
+                            if type_.type.is_fixed_array:  # arr[i][]
+                                val = int(str(type_.type.length))
+                            if isinstance(val, str):  # arr[][i]
+                                # The length of dynamic arrays is stored at the starting slot
+                                val = int(val, 16)
+
+                            elems[i]["elems"] = {}
+                            for j in range(min(val, max_depth)):
+                                slot, value, type_string = get_storage_slot_and_val(
+                                    contracts,
+                                    address,
+                                    var_name,
+                                    rpc_url,
+                                    storage_address,
+                                    key=str(i),
+                                    deep_key=str(j),
+                                )
+                                elems[i]["elems"][j] = {
+                                    "slot": slot,
+                                    "value": value,
+                                    "type_string": type_string,
+                                }
+
+                tmp[var_name]["elems"] = elems
+
+            data[contract] = tmp
+
+    with open(f"{address}_storage_layout.json", "w", encoding="utf-8") as f:
+        json.dump(data, f, indent=4)
+
+
+# pylint: disable=too-many-statements,too-many-branches,inconsistent-return-statements
```

Comment:
Similarly, I think we can split this function into sub-functions to not remove all these warnings